odbc/src/api/functions.rs

```diff
@@ -171,19 +172,80 @@ macro_rules! panic_safe_exec {
         };
     }};
 }
-pub(crate) use panic_safe_exec;
+pub(crate) use panic_safe_exec_clear_diagnostics;
+
+// panic_safe_exec_keep_diagnostics executes `function` such that any panics do not crash the runtime.
+// If a panic occurs during execution, the panic is caught and turned into a String.
+// The panic message is added to the diagnostics of `handle` and SqlReturn::ERROR returned.
+macro_rules! panic_safe_exec_keep_diagnostics {
```

Comment:
I duplicated the macro and just renamed it to make it clear what it was doing rather than adding an argument.